src/cell_lists/spatial_hashing.jl

```diff
@@ -0,0 +1,116 @@
+"""
+    SpatialHashingCellList{NDIMS}(; list_size)
+
+A basic spatial hashing implementation. Similar to [`DictionaryCellList`](@ref), the domain is discretized into cells, 
+and the particles in each cell are stored in a hash map. The hash is computed using the spatial location of each cell 
+[as described by Ihmsen et al. (2001)](@cite Ihmsen2003). By using a hash map, which only stores non-empty cells, 
+the domain is effectively infinite. The size of the hash map is recommended to be approximately twice the number of particles balance memory consumption  
+and the likelihood of hash collisions.
+
+# Arguments
+- `NDIMS::Int`: Number of spatial dimensions (e.g., `2` or `3`).
+- `list_size::Int`: Size of the hash map (e.g., `2 * n_points`) .
+"""
+struct SpatialHashingCellList{CL, CI, CF}
+    list_size::Int
+    NDIMS::Int
+    points::CL
+    coords::CI
+    collisions::CF
+end
+
+@inline index_type(::SpatialHashingCellList) = Int32
+
+function supported_update_strategies(::PointNeighbors.SpatialHashingCellList)
+    return (SerialUpdate, SemiParallelUpdate)
```

Comment:
```suggestion
    return (SerialUpdate)
```
Then you say "this NHS only supports reinitialization", and you don't need the special `update!` anymore.